### PR TITLE
Enable babel on sources

### PR DESCRIPTION
**Pull request by yurydelendik**
First attempt to introduce ES modules. Also replaces esprima by acorn -- webpack is using this dependency. We cannot proceed with mass conversion just yet, see https://github.com/systemjs/systemjs/issues/1630

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -110,13 +110,20 @@ function createWebpackConfig(defines, output) {
     module: {
       loaders: [
         {
+          test: /(\bweb\b).*?\.js$/,
```
Comment (Snuffleupagus) [inline gulpfile.js]: Unless I'm totally misunderstanding the RegExp, won't this limit Babel to files in the `/web` directory? Don't we also want to allow ES6 in files in the `/src` directory too?

Comment (yurydelendik) [inline gulpfile.js]: Sorry forgot to push local branch, fixed.

**Comment by Snuffleupagus** [comment]
> We cannot proceed with mass conversion just yet, see systemjs/systemjs#1630

To avoid being blocked on an upstream issue, which we don't know if/when it will be resolved, in the interest of getting access to ES6 sooner rather than later could we perhaps consider doing the conversion in two steps instead?
Based on your example from https://github.com/yurydelendik/pdf.js/commit/8ef5ddab655ce77e620fd575528719d298fdca20, I think that the following could be used as a stopgap until everything has been converted to ES6 modules:
```javascript
import uiUtils from 'pdfjs-web/ui_utils';
import pdfRenderingQueue from 'pdfjs-web/pdf_rendering_queue';
import pdfjsLib from 'pdfjs-web/pdfjs';
import domEvents from 'pdfjs-web/dom_events';

var { CSS_UNITS, DEFAULT_SCALE, getOutputScale, approximateFraction,
      roundToDivide, RendererType } = uiUtils;
var { RenderingStates } = pdfRenderingQueue;
```

*Please note:* In my testing the `import * as name from 'some/path/to/file';` format doesn't actually works *unless* the module you're importing from is an ES6 module.

**Comment by yurydelendik** [comment]
They problem with ES module dependencies was address.

File: gulpfile.js (hunk 1 above)
Comment (Snuffleupagus) [inline gulpfile.js]: *Really sorry to keep asking, but I'm still not sure that I understand the meaning of this RegExp.*

Don't you want to use e.g. `/\.js$/` instead, such that this is *not* limited to just the `web` folder?

**Review by Snuffleupagus — APPROVED** [review]
I'm very happy to see this work complete, thank you for doing this!

This looks OK to me, with one more inline question answered.

File: gulpfile.js (hunk 1 above)
Comment (yurydelendik) [inline gulpfile.js]: Right, I forgot to change that to add 'src'. On the second thought, enabled for global (not limited to the web folder) will work fine too. Removing the line.

**Comment by timvandermeij** [comment]
Awesome work! I'm also happy that we can move forward with this!

**Pull request merged**